cmake/compiler/arcmwdt/generic.cmake

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Configures CMake for using ccac
+
+find_program(CMAKE_DTS_PREPROCESS_C_COMPILER arc-elf32-gcc)
```

Comment:
CCAC can't be used here as it can't handle DTS files (that's why we create this workaround for metaware toolchain)
I can replace ARC GCC usage with host GCC (however I'd prefer to use ARC one)

Reply:
@ruuddw see internal Jira issue `P10019563-38578` for all the gory details.

Reply:
If the ccac compiler driver is 'too clevel', just use clangac directly?
But, this can be changed later, when the objcopy dependency is resolved as well.

Reply:
Be careful here.
Could be wise to check for both host gcc, as that will most likely be available on Linux machines, however, on Windows, then it might be more likely to find a cross gcc variant.

So probably check for multiple options, and use the best match.
(See also other comment regarding `objcopy` for inspiration.)

Reply:
> CCAC can't be used here as it can't handle DTS files 

Could you elaborate on what issues are seen ?
Maybe @mbolivar-nordic  or @galak would be able to evaluate if DTS files could be made in a more non-GCC friendly way.

Reply:
The main issue is quite simple - MWDT compiler has strict restrictions about file extensions, so it throws error when gets something.dts